### PR TITLE
TEC-11227 extend existing annotations

### PR DESCRIPTION
**Pull request by diepjy**
- Add path to the operation annotation as it's going to be easier to generate the documentation from explicitly providing this n the annotation rathe than from Vert.x.
- Add rate limits as this is outside the scope of the OpenAPI spec.
- Add description annotation to allow response body fields to have an optional description. This is not available in the openAPI spec but it would be nice to have a description for the fields. This would be used in the response class and used within the components part of the documentation.
- Bump the version to upload to Archiva.

OpenAPI spec: https://swagger.io/specification/

### CODE REVIEW

File: modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/Operation.java [hunk 1]
```diff
@@ -60,6 +61,13 @@
 @Retention(RetentionPolicy.RUNTIME)
 @Inherited
 public @interface Operation {
+    /**
+     * The path of the operation.
+     *
+     * @return the path of the operation
+     */
+    String path();
```
Comment (MarcFletcher) [inline modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/Operation.java:69]: Looking at some of the swagger examples they use a [`@Path`](https://docs.oracle.com/javaee/7/api/javax/ws/rs/Path.html) annotation for this:

    @GET
    @Path("/{username}")
    @Operation(summary = "Get user by user name",
            responses = {

Similarly they seem to use the [`@GET`](https://docs.oracle.com/javaee/7/api/javax/ws/rs/GET.html) annotation as well. There's similarly `@POST`, `@HEAD`, `@PUT` etc.

File: modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/ratelimits/RateLimit.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.swagger.v3.oas.annotations.ratelimits;
+
+/**
+ * The annotation may be used to define a custom rate limit
+ */
+public @interface RateLimit {
```
Comment (MarcFletcher) [inline modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/ratelimits/RateLimit.java:22]: If we implemented this as our own annotation, i.e. not inside `@Operation` we wouldn't need to deploy a modified swagger artifact at this point. What do you think?

File: modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/ratelimits/RateLimit.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+   */
+  String name();
+  
+  /**
+   * The maximum number of times the operation can be called within the interval secs. The
+   * default of -1 means there is no rate limit on the operation
+   *
+   * @return the rate limit
+   */
+  int callLimit() default -1;
```
Comment (MarcFletcher) [inline modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/ratelimits/RateLimit.java:37]:
```suggestion
  int maxRequestsInInterval() default -1;
```

File: modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/ratelimits/RateLimit.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  
+  /**
+   * The maximum total number of seconds the operation can be processed. The default of -1 means
+   * there is no rate limit on the operation
+   *
+   * @return the rate limit seconds
+   */
+  int intervalLimitSecs() default -1;
```
Comment (MarcFletcher) [inline modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/ratelimits/RateLimit.java:45]:
```suggestion
  int maxRequestsDurationSecsInInterval() default -1;
```

File: modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/ratelimits/RateLimit.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  /**
+   * The maximum number of times the operation can be called within the interval secs. The
+   * default of -1 means there is no rate limit on the operation
+   *
+   * @return the rate limit
+   */
+  int callLimit() default -1;
+  
+  /**
+   * The maximum total number of seconds the operation can be processed. The default of -1 means
```
Comment (MarcFletcher) [inline modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/ratelimits/RateLimit.java:40]:
```suggestion
   * The maximum runtime seconds of all requests in the interval. The default of -1 means
```

File: modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/ratelimits/RateLimit.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+   * there is no rate limit on the operation
+   *
+   * @return the rate limit seconds
+   */
+  int intervalLimitSecs() default -1;
+  
+  /**
+   * The interval seconds for the entire rate limit on the operation
```
Comment (MarcFletcher) [inline modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/ratelimits/RateLimit.java:48]:
```suggestion
   * The interval duration in seconds
```

**Review by MarcFletcher — CHANGES_REQUESTED** [review]

File: modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/Operation.java (hunk 1 above)
Comment (diepjy) [reply modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/Operation.java:69]: Yes, they're using javax so we can also use that annotation - it shouldn't have any consequences to us during runtime I don't think so this should be safe.

File: modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/ratelimits/RateLimit.java (hunk 2 above)
Comment (diepjy) [reply modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/ratelimits/RateLimit.java:22]: Yes we can move it out of the operation and move this into the CoreAPI.

**Comment by diepjy** [comment]
@MarcFletcher RE: the description annotation - this only exists in the spec as a description for a parameter in the URL and not part of the schema where objects are defined - https://swagger.io/specification/ (Components Object). 
but if we're going to be building our own compoenent objects from the response classes themselves, then we can add our own annotations for our own description field we don't need them in this project, espeically if we're generating our own documentation anyway.

We can probably close this repo if we're going define the custom annotations ourselves.

File: modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/ratelimits/RateLimit.java (hunk 3 above)
Comment (diepjy) [reply modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/ratelimits/RateLimit.java:37]: If we move this to the core API then I'll make the changes in https://github.com/ebx/main/pull/5362

**Comment by diepjy** [comment]
Closing the PR as it's not required - we'll define our own annotations and attemot to use the existnig ones.

**Pull request closed without merging**